### PR TITLE
The table creation function for the package.

### PR DESCRIPTION
**Pull request by voremargot**
I want to consult with Zack before finalizing the function to make sure they work in a similar way.  Please do an initial run through of the script to highlight any changes I might need to makes! Thanks!

### CODE REVIEW

File: src/straPy/data_vis.py [hunk 1]
```diff
@@ -1,58 +1,140 @@
...
-        bootstrapped sampling distribution
-    ci : numpy.ndarray
-        confidence interval
-    size : boolean, default=True
-        sample size
-    mean: boolean, default=True
-        sample mean
-    alpha : float, default=0.05
-        selected alpha
+    stat_dic : dict
```
Comment (gauthampughazhendhi) [inline src/straPy/data_vis.py]:
```suggestion
    stats : dict
```
Just suggesting this name for uniformity.

File: src/straPy/data_vis.py [hunk 2]
```diff
@@ -1,58 +1,140 @@
...
+        include the significance level in the summary statistics table
 
     Returns
     -------
-    table object
-        table with lower bound and upper bound of the confidence interval and relevant statistics
+    summary statistics: table object
+        table summerizing the lower bound and upper bound of the confidence interval, the standard error, the sampling statitic (if estimator = True), and the significance level (if alpha = True)
+    bootstrap parameters: table object
+        table  summerizing the parameters of the bootstrap sampling spficiying the original sample size, number of repititions, the significance level, and the number of samples in each bootstrap if its different from the original sample size.
```
Comment (gauthampughazhendhi) [inline src/straPy/data_vis.py]: According to Python's pep8 styling guide, the number of characters should not be more than 79 characters. Can you break these lines?

**Review by gauthampughazhendhi — CHANGES_REQUESTED** [review]
@voremargot the function looks good to me, I have added a few minor changes. Also, when you are done with the changes, can you add an image of the final table image to this PR conversation? Just to have a look at the table. Thanks for the good work!

File: src/straPy/data_vis.py [hunk 3]
```diff
@@ -1,58 +1,140 @@
...
     -------
-    plot object
-        density plot with confidence interval
+    plot: histogram
+        histogram of bootstrapped distribution with confidence interval and observed mean
     
     Examples
     --------
-    >>> density_ci_plot([1, 1, 2, 3, 5, 10]，[1., 9.375]）
+    >>> histogram_ci_plot([1, 2, 3, 4, 5, 6, 7], 1000 ,n=100, ci_level=0.95, ci_random_seed=123)
```
Comment (BooleanJulien) [inline src/straPy/data_vis.py:39]: We will need to remember to cascade this change to the readme

File: src/straPy/data_vis.py [hunk 4]
```diff
@@ -1,58 +1,140 @@
...
-        confidence interval
-    size : boolean, default=True
-        sample size
-    mean: boolean, default=True
-        sample mean
-    alpha : float, default=0.05
-        selected alpha
+    stat_dic : dict
+        summary statistics produced by the `calculate_boot_stats()` function 
+    precision : int, default= 2
```
Comment (BooleanJulien) [inline src/straPy/data_vis.py]:
```suggestion
    precision : int, default=2
```

File: src/straPy/data_vis.py [hunk 5]
```diff
@@ -1,58 +1,140 @@
...
+        the precision of the table values
+    esitmator : boolean, default=True
```
Comment (BooleanJulien) [inline src/straPy/data_vis.py]:
```suggestion
    estimator : boolean, default=True
```

File: src/straPy/data_vis.py [hunk 6]
```diff
@@ -1,58 +1,140 @@
...
\ No newline at end of file
+    if not isinstance(stat_dic, dict):
+        raise TypeError("Input statistics must be a dictionary")
+            
+    if not isinstance(precision, int):
+        raise TypeError("The precision parameter must be of type int.")
+            
+    if not (isinstance(estimator, bool) or 
+            isinstance(alpha, bool)):
+        raise TypeError("The estimator and alpha parameter  must be of type boolean.")
```
Comment (BooleanJulien) [inline src/straPy/data_vis.py]:
```suggestion
        raise TypeError("The estimator and alpha parameter must be of type boolean.")
```

File: src/straPy/data_vis.py [hunk 7]
```diff
@@ -1,58 +1,140 @@
...
+    
+    # define the statistics table
+    df = pd.DataFrame(data=np.array([(stat_dic["lower"], stat_dic["upper"], stat_dic["std_err"])]),
+                      columns=["Lower bound CI", "Upper bound CI", "Standard Error"])
```
Comment (BooleanJulien) [inline src/straPy/data_vis.py]:
```suggestion
                      columns=["Lower Bound CI", "Upper Bound CI", "Standard Error"])
```

File: src/straPy/data_vis.py [hunk 8]
```diff
@@ -1,58 +1,140 @@
...
+
+    if alpha is True:
+        df["Significance Level"] = 1 - stat_dic["level"]
+        stats_table = df.style.format(
+            precision=precision, formatter={("Significance Level"): "{:.3f}"}
+        )
+    else:
+        stats_table = df.style.format(precision=precision)
+
+    #set formatting and caption for table
```
Comment (BooleanJulien) [inline src/straPy/data_vis.py]:
```suggestion
    # set formatting and caption for table
```

File: src/straPy/data_vis.py [hunk 9]
```diff
@@ -1,58 +1,140 @@
...
+    #create bootstrapping parameter summary table
+    df_bs = pd.DataFrame(
+        data=np.array(
+            [(stat_dic["sample_size"], stat_dic["rep"], (1 - stat_dic["level"]))]),
+        columns=["Sample Size", "Repetition", "Significance Level"])
+    
+    if stat_dic["n"] != "auto":
+        df_bs["Samples per bootstrap"] = round(stat_dic["n"], 0)
+
+    #set formatting and caption for table
```
Comment (BooleanJulien) [inline src/straPy/data_vis.py]:
```suggestion
    # set formatting and caption for table
```

File: src/straPy/data_vis.py [hunk 10]
```diff
@@ -1,58 +1,140 @@
...
+    stats_table.set_caption(
+        "Bootstrapping sample statistics from sample with "+ 
+        str(stat_dic["sample_size"]) + " records"
+    ).set_table_styles(
+        [{"selector": "caption", "props": "caption-side: bottom; font-size:1.00em;"}],
+        overwrite=False)
+
+    #create bootstrapping parameter summary table
```
Comment (BooleanJulien) [inline src/straPy/data_vis.py]:
```suggestion
    # create bootstrapping parameter summary table
```

**Review by BooleanJulien — APPROVED** [review]
Some minor formatting suggestions, overall LGTM!

Echoing Gautham's comments on providing some examples, hard to properly review the table creation without seeing output, but I'm sure we will get to that later. Good work! :)

File: src/straPy/data_vis.py (hunk 2 above)
Comment (voremargot) [reply src/straPy/data_vis.py]: This has been updated

**Comment by voremargot** [comment]
This is an image of the stat table output:
image

**Comment by voremargot** [comment]
This is an image of the parameter table output:
image

**Comment by voremargot** [comment]
@gauthampughazhendhi @BooleanJulien  Please take another look at the function! I have made all the suggested changes and have caught a couple extra little errors. Thanks for the great review!

**Comment by voremargot** [comment]
Please check my test function script!

**Comment by gauthampughazhendhi** [comment]
@voremargot everything looks good to me, excellent work!
1. I can see the notebook is still part of the PR, is that intentional?
2. Also, the code lines are still above 79 chars, but I saw your comment saying it's changed now.

**Review by gauthampughazhendhi — APPROVED** [review]
Looks good to me. Thanks for the work!

**Pull request merged**